tests/test_revision.py

```diff
@@ -1213,3 +1254,239 @@ def test_failure_message(self):
         assert_raises_message(
             RevisionError, "Dependency resolution failed;", list, iter_
         )
+
+
+class InvalidRevisionMapTest(TestBase):
+    def _assert_raises_revision_map(self, map_, except_cls, msg):
+        try:
+            map_._revision_map
+            assert False, "revision_map is successfully loaded"
+        except except_cls as e:
+            assert re.search(
+                msg, util.text_type(e), re.UNICODE
+            ), "%r !~ %s" % (msg, e)
+            print(util.text_type(e).encode("utf-8"))
+
+    def _assert_raises_revision_map_loop(self, map_):
+        self._assert_raises_revision_map(
+            map_,
+            LoopDetected,
+            "Self-loop is detected",
```

Comment:
**mike bayer** (zzzeek) wrote:

the message actually being raised here is: "Self-loop is detected Self-loop is detected" due to the mis-use in revision.py.   so let's add an argument here and elsewhere to assert the whole message; here's part of it:

--- a/tests/test_revision.py
    +++ b/tests/test_revision.py
    @@ -1267,11 +1267,11 @@ class InvalidRevisionMapTest(TestBase):
                 ), "%r !~ %s" % (msg, e)
                 print(util.text_type(e).encode("utf-8"))
     
    -    def _assert_raises_revision_map_loop(self, map_):
    +    def _assert_raises_revision_map_loop(self, map_, revision):
             self._assert_raises_revision_map(
                 map_,
                 LoopDetected,
    -            "Self-loop is detected",
    +            r"^Self-loop is detected on %s$" % revision,
             )
     
         def _assert_raises_revision_map_dep_loop(self, map_):
    @@ -1303,7 +1303,7 @@ class GraphWithLoopTest(InvalidRevisionMapTest):
                     Revision("a", "a"),
                 ]
             )
    -        self._assert_raises_revision_map_loop(map_)
    +        self._assert_raises_revision_map_loop(map_, "a")
     
         def test_revision_map_base_loop(self):
             map_ = RevisionMap(
    @@ -1313,7 +1313,7 @@ class GraphWithLoopTest(InvalidRevisionMapTest):
                     Revision("c", "b"),
                 ]
             )
    -        self._assert_raises_revision_map_loop(map_)
    +        self._assert_raises_revision_map_loop(map_, "a")
     
         def test_revision_map_head_loop(self):
             map_ = RevisionMap(
    @@ -1323,7 +1323,7 @@ class GraphWithLoopTest(InvalidRevisionMapTest):
                     Revision("c", ("b", "c")),
                 ]
             )
    -        self._assert_raises_revision_map_loop(map_)
    +        self._assert_raises_revision_map_loop(map_, "c")
     
         def test_revision_map_branch_point_loop(self):
             map_ = RevisionMap(
    @@ -1334,7 +1334,7 @@ class GraphWithLoopTest(InvalidRevisionMapTest):
                     Revision("c2", "b"),
                 ]
             )
    -        self._assert_raises_revision_map_loop(map_)
    +        self._assert_raises_revision_map_loop(map_, "b")
     
         def test_revision_map_merge_point_loop(self):
             map_ = RevisionMap(
    @@ -1345,7 +1345,7 @@ class GraphWithLoopTest(InvalidRevisionMapTest):
                     Revision("c", ("b1", "b2", "c")),
                 ]
             )
    -        self._assert_raises_revision_map_loop(map_)
    +        self._assert_raises_revision_map_loop(map_, "c")
     
         def test_revision_map_solitary_dependency_loop(self):
             map_ = RevisionMap(

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2379